sql/driver/rows.go

```diff
@@ -43,6 +58,8 @@ func (r *rows) Next(dest []driver.Value) error {
 	if r.pos >= len(r.rows) {
 		return io.EOF
 	}
-	dest[0] = r.rows[r.pos]
+	for i, v := range r.rows[r.pos] {
```

Comment:
`len(dest)` is guaranteed to be `len(r.Columns())`: https://golang.org/pkg/database/sql/driver/#Rows

        // Next is called to populate the next row of data into
        // the provided slice. The provided slice will be the same
        // size as the Columns() are wide.
        //
        // The dest slice may be populated only with
        // a driver Value type, but excluding string.
        // All string values must be converted to []byte.
        //
        // Next should return io.EOF when there are no more rows.